src/main/java/by/itacademy/javaenterprise/goralchuk/dao/implementation/PeopleDaoImpl.java

```diff
@@ -44,21 +44,80 @@ public boolean save(People people) {
 
     @Override
     public boolean update(People peopleNew){
+        if (peopleNew != null) {
+            try {
+                People peopleOld = em.find(People.class, peopleNew.getId());
+                try {
+                    em.detach(peopleOld);
+                    if (peopleNew.getName() != null){
+                        peopleOld.setName(peopleNew.getName());
+                    }
+                    if (peopleNew.getSurname() != null){
+                        peopleOld.setSurname(peopleNew.getSurname());
+                    }
+                    if (peopleNew.getPetPeople() != null){
+                        peopleOld.setPetPeople(peopleNew.getPetPeople());
+                    }
+                    em.getTransaction().begin();
+                    em.merge(peopleOld);
+                    em.getTransaction().commit();
+                    return true;
+                } catch (Exception e) {
+                    em.getTransaction().rollback();
+                    logger.error("Transaction failed " + e.getMessage(), e);
+                    return false;
+                }
+            } catch (IllegalArgumentException ex) {
+                logger.error("No such object was found in the database", ex);
+            }
+        }
+        logger.error(new NullPointerException("Method field cannot be empty").getMessage());
```

Comment:
Зачем создавать эксепшен? просто напиши сообщение с ошибкой. 
```suggestion
        logger.error("Method field cannot be empty");
```

-